src/lava/proc/graded/models.py

```diff
@@ -0,0 +1,152 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+import numpy as np
+
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.magma.core.model.py.ports import PyInPort, PyOutPort
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.resources import CPU
+from lava.magma.core.decorator import implements, requires, tag
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+
+from lava.proc.graded.process import GradedVec, NormVecDelay, InvSqrt
+
+
+class AbstractGradedVecModel(PyLoihiProcessModel):
+    """Implementation of GradedVec"""
+
+    a_in = None
```

Comment:
Same as in other PR: This should be LavaPyTypes and in the specific implementation of the Abstract class, they do not need to be added again.

Reply:
Fix it later